pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1600,7 +1667,14 @@ public void closeFailed(ManagedLedgerException exception, Object ctx) {
             closeFuture.completeExceptionally(exception);
             return null;
         });
-
+        if (!disconnectClients) {
+            FutureUtil.completeAfter(closeFutures.notDisconnectClients, closeFuture);
+        } else {
+            FutureUtil.completeAfter(closeFutures.notDisconnectClients, closeFuture);
+            FutureUtil.completeAfter(closeFutures.notWaitDisconnectClients, closeFuture);
+            FutureUtil.completeAfterAll(closeFutures.waitDisconnectClients, closeFuture,
+                    FutureUtil.waitForAll(futures));
```

Comment:
Why do we wait for futures here? Dont we already have this logic handled at line 1639?

Reply:
If `closeWithoutWaitingClientDisconnect` is `true`, `clientCloseFuture` is not handled `futures`

```java
CompletableFuture<Void> clientCloseFuture = closeWithoutWaitingClientDisconnect
                ? CompletableFuture.completedFuture(null)
                : FutureUtil.waitForAll(futures);
```

Reply:
if (closeWithoutWaitingClientDisconnect) {
  clientCloseFuture =  CompletableFuture.completedFuture(null);
}

so it will not wait for `futures `

Reply:
I am sorry.  I think I haven't clarified "waiting for this futures" in my above comment.


Here, even if `closeWithoutWaitingClientDisconnect = true`, we are unnecessarily binding `FutureUtil.waitForAll(futures)` to  `closeFuture` by `FutureUtil.completeAfterAll(closeFutures.waitDisconnectClients, closeFuture,
                    FutureUtil.waitForAll(futures));`. This will increase the overhead of topic.close from `auto bunde unloading`. 

nit: Ideally, we shouldn't populate these `futures` when `closeWithoutWaitingClientDisconnect=false`. This overhead can add up when we many producers and consumers for each topic.

Reply:
Ah, I think you are right. Improved. And we changed the behavior when `closeWithoutWaitingClientDisconnect = true`: 

**Before**
- disconnect clients
  - if failed, call `unfenceTopicToResume` and do not close Managed Ledger.

**After**
- disconnect clients
- close Managed Ledger.